Sources/Amplitude/AMPStorage.m

```diff
@@ -124,19 +110,22 @@ + (NSMutableArray *)getEventsFromDisk:(NSString *)path {
 }
 
 + (NSDictionary *)JSONFromFile:(NSString *)path {
-    NSData *data = [NSData dataWithContentsOfFile:path];
-    if (data == nil) {
-        NSString *emptyData = @"{ \"batch\": []}";
-        data = [emptyData dataUsingEncoding:NSUTF8StringEncoding];
+    NSMutableString *content = [NSMutableString stringWithContentsOfFile:path encoding:NSUTF8StringEncoding error:nil];
+    if (content == nil) {
+        content = [[NSMutableString alloc] init];
+        [content appendString:@"{ \"batch\": ["];
     }
 
-    NSError *err = nil;
-    NSDictionary *jsonAsDict = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:&err];
-    if (err != nil || jsonAsDict == nil) {
-        [AMPStorage finish:path];
-        data = [NSData dataWithContentsOfFile:path];
-        jsonAsDict = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:&err];
+    [content appendString:@"]}"];
+    NSData *completeData = [content dataUsingEncoding:NSUTF8StringEncoding];
+    NSDictionary *jsonAsDict = [NSJSONSerialization JSONObjectWithData:completeData options:kNilOptions error:nil];
```

Comment:
How's the performance on this way comparing to the old way?

Reply:
I tried to add 1000 events into the file storage and call getEventsFromDisk once on different methods. The execute time is similar. 
I also tried to execute storeEvent -> getEventsFromDisk -> remove for 100 times, the new way win.